File: Programmers/Level3/security_camera/SolutionTest.java
package security_camera;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.Test;

public class SolutionTest {

	private Solution s = new Solution();
	
	@Test
	public void case1() {
		int[][] routes = {{-20,15}, {-14,-5}, {-18,-13}, {-5,-3}};
		assertEquals(2, s.solution(routes));
	}
	
	@Test
	public void case2() {
		int[][] routes = {{0,0}, {0,0}, {2,2}};
		assertEquals(2, s.solution(routes));
	}
	
	@Test
	public void case3() {
		int[][] routes = {{0,1}, {0,1}, {2,2}};
		assertEquals(2, s.solution(routes));
	}
	
}

File: src/main/java/programmers/kakao2023/expressionable_binary_trees/SolutionTest.kt
package programmers.kakao2023.expressionable_binary_trees

import org.junit.jupiter.api.Test
import kotlin.test.assertContentEquals

class SolutionTest {

    val solution = Solution()

    @Test
    fun case1() {
        val numbers = longArrayOf(7, 42, 5)
        val expected = intArrayOf(1, 1, 0)
        val actual = solution.solution(numbers)
        assertContentEquals(expected, actual)
    }

    @Test
    fun case2() {
        val numbers = longArrayOf(63, 111, 95)
        val expected = intArrayOf(1, 1, 0)
        val actual = solution.solution(numbers)
        assertContentEquals(expected, actual)
    }

    @Test
    fun case3() {
        val numbers = longArrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 128, 129, 16512, 2147516555)
        val expected = intArrayOf(1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1)
        val actual = solution.solution(numbers)
        assertContentEquals(expected, actual)
    }

}